File: old_samples/sample-videochat-conference/src/main/java/com/quickblox/sample/conference/adapters/DialogsAdapter.java
package com.quickblox.sample.conference.adapters;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.quickblox.chat.model.QBChatDialog;
import com.quickblox.sample.conference.R;
import com.quickblox.sample.core.ui.adapter.BaseSelectableListAdapter;
import com.quickblox.sample.core.utils.ResourceUtils;
import com.quickblox.sample.core.utils.UiUtils;

import java.util.List;


public class DialogsAdapter extends BaseSelectableListAdapter<QBChatDialog> {

    public DialogsAdapter(Context context, List<QBChatDialog> dialogs) {
        super(context, dialogs);
    }

    @Override
    public View getView(final int position, View convertView, ViewGroup parent) {
        ViewHolder holder;
        if (convertView == null) {
            convertView = inflater.inflate(R.layout.list_item_dialog, parent, false);

            holder = new ViewHolder();
            holder.rootLayout = (ViewGroup) convertView.findViewById(R.id.root);
            holder.nameTextView = (TextView) convertView.findViewById(R.id.text_dialog_name);
            holder.dialogImageView = (ImageView) convertView.findViewById(R.id.image_dialog_icon);

            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }

        QBChatDialog dialog = getItem(position);
        holder.dialogImageView.setBackgroundDrawable(UiUtils.getGreyCircleDrawable());

        holder.nameTextView.setText(dialog.getName());

        holder.rootLayout.setBackgroundColor(isItemSelected(position) ? ResourceUtils.getColor(R.color.selected_list_item_color) :
                ResourceUtils.getColor(android.R.color.transparent));


        return convertView;
    }

    public void toggleOneItem(QBChatDialog item) {
        if (selectedItems.contains(item)) {
            selectedItems.remove(item);
        } else {
            selectedItems.clear();
            selectedItems.add(item);
        }
        notifyDataSetChanged();
    }

    private static class ViewHolder {
        ViewGroup rootLayout;
        ImageView dialogImageView;
        TextView nameTextView;
    }
}

File: sample-videochat-conference-kotlin/src/main/java/com/quickblox/sample/videochat/conference/kotlin/adapter/DialogsAdapter.kt
package com.quickblox.sample.videochat.conference.kotlin.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.quickblox.chat.model.QBChatDialog
import com.quickblox.sample.videochat.conference.kotlin.R
import com.quickblox.sample.videochat.conference.kotlin.utils.getColor
import com.quickblox.sample.videochat.conference.kotlin.utils.getGreyCircleDrawable
import kotlinx.android.synthetic.main.list_item_dialog.view.*


class DialogsAdapter(val context: Context,
                     private var items: List<QBChatDialog>,
                     private val clickListener: OnItemClickListener) : RecyclerView.Adapter<DialogsAdapter.ViewHolder>() {

    private val selectedItems: MutableList<QBChatDialog> = ArrayList()

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.nameTextView.text = items[position].name
        holder.dialogImageView.setBackgroundDrawable(getGreyCircleDrawable())

        val colorResource = if (isItemSelected(items[position])) {
            getColor(R.color.selected_list_item_color)
        } else {
            getColor(android.R.color.transparent)
        }

        holder.rootLayout.setBackgroundColor(colorResource)
        holder.rootLayout.setOnClickListener { clickListener.onShortClick(items[position]) }
        holder.rootLayout.setOnLongClickListener { clickListener.onLongClick(items[position]);true }
    }

    override fun getItemCount(): Int {
        return items.size
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(context).inflate(R.layout.list_item_dialog, parent, false))
    }

    fun updateDialogs(items: List<QBChatDialog>) {
        this.items = items
        notifyDataSetChanged()
    }

    private fun isItemSelected(item: QBChatDialog): Boolean {
        return selectedItems.contains(item)
    }

    fun getSelectedItems(): List<QBChatDialog> {
        return selectedItems
    }

    fun clearSelectedItems() {
        selectedItems.clear()
        notifyDataSetChanged()
    }

    fun selectItem(dialog: QBChatDialog) {
        if (selectedItems.contains(dialog)) {
            return
        }
        selectedItems.add(dialog)
        notifyDataSetChanged()
    }

    fun toggleSelection(dialog: QBChatDialog) {
        if (selectedItems.contains(dialog)) {
            selectedItems.remove(dialog)
        } else {
            selectedItems.add(dialog)
        }
        notifyDataSetChanged()
    }

    fun toggleOneItem(dialog: QBChatDialog) {
        if (selectedItems.contains(dialog)) {
            selectedItems.remove(dialog)
        } else {
            selectedItems.clear()
            selectedItems.add(dialog)
        }
        notifyDataSetChanged()
    }

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val nameTextView = view.text_dialog_name
        val dialogImageView = view.image_dialog_icon
        val rootLayout = view.root
    }

    interface OnItemClickListener {
        fun onShortClick(dialog: QBChatDialog)

        fun onLongClick(dialog: QBChatDialog)
    }
}